File: apps/zxtune-android/zxtune/src/main/java/app/zxtune/fs/http/HttpObject.java
package app.zxtune.fs.http;

import android.net.Uri;

import androidx.annotation.Nullable;

import java.io.IOException;
import java.io.InputStream;

import app.zxtune.TimeStamp;

public interface HttpObject {

    Uri getUri();

    @Nullable
    Long getContentLength();

    @Nullable
    TimeStamp getLastModified();

    InputStream getInput() throws IOException;
}

File: apps/zxtune-android/zxtune/src/main/java/app/zxtune/fs/http/HttpObject.kt
package app.zxtune.fs.http

import android.net.Uri
import app.zxtune.TimeStamp
import java.io.IOException
import java.io.InputStream

interface HttpObject {

    val uri: Uri

    val contentLength: Long?

    val lastModified: TimeStamp?

    @get:Throws(IOException::class)
    val input: InputStream
}
